toxcore/group_moderation.c

```diff
@@ -220,10 +227,10 @@ int mod_list_add_entry(GC_Chat *chat, const uint8_t *mod_data)
 
     uint8_t **tmp_list = realloc(chat->moderation.mod_list, sizeof(uint8_t *) * (chat->moderation.num_mods + 1));
 
-    if (tmp_list == NULL) {
-        chat->moderation.num_mods = 0;
+    if (tmp_list == NULL)
         return -1;
-    }
+
+    chat->moderation.mod_list = tmp_list;
```

Comment:
No, I think. mod_list just resized, but not initialized and num_mods not chaged. Next call for realloc with same size is safe.